tests/bot/cogs/test_token_remover.py

```diff
@@ -1,131 +1,279 @@
-import asyncio
-import logging
 import unittest
-from unittest.mock import AsyncMock, MagicMock
+from unittest import mock
+from unittest.mock import MagicMock
 
 from discord import Colour
 
-from bot.cogs.token_remover import (
-    DELETION_MESSAGE_TEMPLATE,
-    TokenRemover,
-    setup as setup_cog,
-)
-from bot.constants import Channels, Colours, Event, Icons
-from tests.helpers import MockBot, MockMessage
+from bot import constants
+from bot.cogs import token_remover
+from bot.cogs.moderation import ModLog
+from bot.cogs.token_remover import TokenRemover
+from tests.helpers import MockBot, MockMessage, autospec
 
 
-class TokenRemoverTests(unittest.TestCase):
+class TokenRemoverTests(unittest.IsolatedAsyncioTestCase):
     """Tests the `TokenRemover` cog."""
 
     def setUp(self):
         """Adds the cog, a bot, and a message to the instance for usage in tests."""
         self.bot = MockBot()
-        self.bot.get_cog.return_value = MagicMock()
-        self.bot.get_cog.return_value.send_log_message = AsyncMock()
         self.cog = TokenRemover(bot=self.bot)
 
-        self.msg = MockMessage(id=555, content='')
-        self.msg.author.__str__ = MagicMock()
-        self.msg.author.__str__.return_value = 'lemon'
-        self.msg.author.bot = False
-        self.msg.author.avatar_url_as.return_value = 'picture-lemon.png'
-        self.msg.author.id = 42
-        self.msg.author.mention = '@lemon'
+        self.msg = MockMessage(id=555, content="hello world")
         self.msg.channel.mention = "#lemonade-stand"
+        self.msg.author.__str__ = MagicMock(return_value=self.msg.author.name)
+        self.msg.author.avatar_url_as.return_value = "picture-lemon.png"
 
-    def test_is_valid_user_id_is_true_for_numeric_content(self):
-        """A string decoding to numeric characters is a valid user ID."""
-        # MTIz = base64(123)
-        self.assertTrue(TokenRemover.is_valid_user_id('MTIz'))
+    def test_is_valid_user_id(self):
+        """Should correctly discern valid user IDs and ignore non-numeric and non-ASCII IDs."""
+        subtests = (
+            ("MTIz", True),  # base64(123)
+            ("YWJj", False),  # base64(abc)
+            ("λδµ", False),
+        )
```

Comment:
I think we should expand upon the subtests to include user IDs that follow a more realistic format, perhaps using the real user IDs of a few staff members (since it's fully public information). Also, having something around 5 valid user IDs and 5 invalid ones would help to improve coverage (not literal code coverage %; I'm referring to coverage of possible inputs).

IMO, the ones might be a bit too unrealistic for testing purposes.

Reply:
I can get 5 valid ones but I don't know would be considered a "realistic" invalid value.

Reply:
That's fine. When I mentioned realistic inputs, I was mainly referring to the valid ones. For the invalid inputs, they should just mainly have some variety.